lib/OpenQA/Controller/Running.pm

```diff
@@ -39,11 +40,9 @@ sub init {
 
     my $basepath = running_log($testdirname);
     $self->stash('basepath', $basepath);
-    my $workerid = $job->worker_id;
-    $self->stash('workerid', $workerid);
-    my $worker = OpenQA::Scheduler::worker_get($workerid);
-    my $workerport = $worker->{properties}->{WORKER_PORT};
-    my $workerurl = $worker->{properties}->{WORKER_IP} . ':' . $workerport;
+    $self->stash('worker', $job->worker);
+    my $workerport = $job->worker->get_property('WORKER_PORT');
+    my $workerurl = $job->worker->get_property('WORKER_IP') . ':' . $workerport;
```

Comment:
depends on your definition of needing:

<pre>
./templates/test/running.html.ep:              <td>Worker: <%= $worker->{properties}->{WORKER_IP} %>:<%= $worker->{properties}->{WORKER_PORT} %></td>
</pre>